q2_fmt/_peds.py

```diff
@@ -170,24 +186,43 @@ def _check_for_time_column(metadata, time_column):
     try:
         num_timepoints = metadata[time_column].dropna().unique().size
     except Exception as e:
-        raise KeyError('There was an error finding %s in the metadata'
-                       % time_column) from e
+        if time_column == metadata.index.name:
+            raise KeyError('The `--p-time-column` input provided was the same'
+                           ' as the index of the metadata. `--p-time-column`'
+                           ' can not be the same as the index of metadata:'
+                           ' `%s`' % time_column) from e
+        else:
+            raise KeyError('There was an error finding the provided'
+                           ' `--p-time-column`: `%s` in the metadata'
+                           % time_column) from e
     return num_timepoints
 
 
-def _check_time_column_numeric(metadata, time_column):
+def _check_time_column_numeric(column_properties, time_column):
     try:
-        metadata[time_column].dropna().astype(int)
-    except Exception as e:
-        raise ValueError('%s must be numeric' % time_column) from e
+        assert column_properties[time_column].type == 'numeric'
+    except AssertionError as e:
+        raise AssertionError('Non-numeric values found in `--p-time-column`.'
+                             ' Please make sure the column selected contains'
+                             ' only simple integer values. Column with'
+                             ' non-numeric values that was'
+                             ' selected: %s' % time_column) from e
 
 
 def _check_reference_column(metadata, reference_column):
     try:
         reference_series = metadata[reference_column]
     except Exception as e:
-        raise KeyError('There was an error finding %s in the metadata'
-                       % reference_column) from e
+        if reference_column == metadata.index.name:
```

Comment:
You could also refactor this branch and the `subject_column` one to be another helper that checks if the given column is equal to the `metadata.index.name`.